### PR TITLE
Wip cmake spdk

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

**Comment by tchaikov** [comment]
@yuyuyu101 mind testing it? thanks.

**Comment by yuyuyu101** [comment]
LGTM

**Comment by liewegas** [comment]
This still requires an out-of-tree spdk source tree.  Would it be better to use a submodule here?

**Comment by tchaikov** [comment]
@yuyuyu101 what do you think?

**Comment by yuyuyu101** [comment]
@liewegas it would be great, plz help to fork from https://github.com/spdk/spdk .

I would like to cooperate with @tchaikov to setup submodule with cmake/automake.

**Comment by liewegas** [comment]
forked!

**Comment by tchaikov** [comment]
moved the KernelDevice.cc change to #7507

**Comment by tchaikov** [comment]
@liewegas @yuyuyu101 done.

**Comment by tchaikov** [comment]
@cbodley could you help take a look at the cmake changes in this pr? thanks. the NVMEDevice change has been reviewed by @yuyuyu101 .

**Comment by cbodley** [comment]
@tchaikov looks good. i like how you did the foreach :+1:

**Comment by tchaikov** [comment]
- added the autotools support
- and include the spdk source the the dist tarball

File: cmake/modules/Finddpdk.cmake [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#
+# DPDK_FOUND
+# DPDK_INCLUDE_DIR
+# DPDK_LIBRARIES
+
+find_path(DPDK_INCLUDE_DIR rte_config.h
+  PATH_SUFFIXES dpdk)
+find_library(DPDK_rte_eal_LIBRARY rte_eal)
+find_library(DPDK_rte_mempool_LIBRARY rte_mempool)
+find_library(DPDK_rte_ring_LIBRARY rte_ring)
```
Comment (yuyuyu101) [inline cmake/modules/Finddpdk.cmake]: I think we could add custom dpdk lib directory:
if (EXISTS ${DPDK_TARGET})
  list(APPEND _dpdk_include_path "${DPDK_TARGET}/include")
  list(APPEND _dpdk_lib_path "${DPDK_TARGET}/lib")
endif()

find_path(DPDK_INCLUDE_DIR rte_eal.h NO_DEFAULT_PATH PATHS ${_dpdk_include_path})
find_library(DPDK_LIBRARY NO_DEFAULT_PATH NAMES  rte_eal PATHS ${_dpdk_lib_path})

**Comment by tchaikov** [comment]
@yuyuyu101 helped tested it. both automake and cmake build with spdk enabled.

**Pull request merged**